pkg/api/v1alpha1/awsiamconfig_webhook.go

```diff
@@ -78,7 +78,7 @@ func validateImmutableAWSIamFields(new, old *AWSIamConfig) field.ErrorList {
 	if !new.Spec.Equal(&old.Spec) {
 		allErrs = append(
 			allErrs,
-			field.Invalid(field.NewPath("spec", "AWSIamConfig"), new, "config is immutable"),
+			field.Forbidden(field.NewPath("AWSIamConfig", new.Name), "config is immutable"),
```

Comment:
What is the logic we follow to define the path?
It seems like we are removing spec from all of them, was that redundant?
Also, why is `new.Name` part of the path?